src/engine/strat_engine/engine.rs

```diff
@@ -113,12 +118,29 @@ impl Engine for StratEngine {
         let pool = StratPool::initialize(name, &dm, blockdev_paths, redundancy, force)?;
 
         let uuid = pool.uuid();
+        devlinks::pool_added(pool.name());
         self.pools.insert(pool);
+
         Ok(uuid)
     }
 
     fn destroy_pool(&mut self, uuid: PoolUuid) -> EngineResult<bool> {
-        destroy_pool!{self; uuid}
+        if let Some(pool) = self.pools.get_by_uuid(uuid) {
```

Comment:
I think you can still use the macro, almost unmodified, so it makes sense to do so.

Reply:
I tried but didn't have any luck without causing other ugly ripples.